dll/win32/shell32/CCopyMoveToMenu.cpp

```diff
@@ -436,3 +436,145 @@ CCopyMoveToMenu::GetSite(REFIID riid, void **ppvSite)
 
     return m_pSite->QueryInterface(riid, ppvSite);
 }
+
+static DWORD
+SetClipboard(UINT cf, const void* data, SIZE_T size)
+{
+    BOOL succ = FALSE;
+    HGLOBAL handle = GlobalAlloc(GMEM_MOVEABLE | GMEM_SHARE, size);
+    if (handle)
+    {
+        LPVOID clipdata = GlobalLock(handle);
+        if (clipdata)
+        {
+            CopyMemory(clipdata, data, size);
+            GlobalUnlock(handle);
+            if (OpenClipboard(NULL))
+            {
+                EmptyClipboard();
+                succ = SetClipboardData(cf, handle) != NULL;
+                CloseClipboard();
+            }
+        }
+        if (!succ)
+        {
+            GlobalFree(handle);
+        }
+    }
+    return succ ? ERROR_SUCCESS : GetLastError();
+}
+
+static DWORD
+SetClipboardFromString(LPCWSTR str)
+{
+    SIZE_T cch = lstrlenW(str) + 1, size = cch * sizeof(WCHAR);
+    if (size > cch)
```

Comment:
What's the idea behind this? For sure, cch * 2 (which is == size) will always be > cch .
Or is it somehow in order to check some sort of value wrapping?

Reply:
The error message returned in the else already answers this.